packages/solana-rpc-provider/lib/SolanaRpcProvider.ts

```diff
@@ -0,0 +1,226 @@
+import { NodeProvider as NodeProvider } from '@liquality/node-provider'
+import { BigNumber, ChainProvider, Address, Block, Transaction, solana } from '@liquality/types'
+import { SolanaNetwork } from '@liquality/solana-network'
+import { TxNotFoundError } from '@liquality/errors'
+
+import filter from 'lodash/filter'
+import Bytecode from './bytecode'
+
+import {
+  Connection,
+  PublicKey,
+  ParsedConfirmedTransaction,
+  AccountInfo,
+  sendAndConfirmTransaction,
+  Transaction as SolTransaction,
+  Keypair,
+  BpfLoader,
+  BPF_LOADER_PROGRAM_ID,
+  ConfirmedSignatureInfo
+} from '@solana/web3.js'
+
+export default class SolanaRpcProvider extends NodeProvider implements Partial<ChainProvider> {
+  _network: SolanaNetwork
+  connection: Connection
+
+  constructor(network: SolanaNetwork) {
+    super({
+      baseURL: network.helperUrl,
+      responseType: 'text',
+      transformResponse: undefined
+    })
+    this.connection = new Connection(network.nodeUrl)
+    this._network = network
+  }
+
+  async generateBlock(numberOfBlocks: number): Promise<void> {
+    await new Promise((resolve) => setTimeout(resolve, numberOfBlocks * 20000))
+  }
+
+  // getBlockByHash(blockHash: string, includeTx?: boolean): Promise<Block<any>> {
```

Comment:
What about this one?

Reply:
Solana do not support get block by hash, just block by number